Assets/VRTK/Scripts/Locomotion/VRTK_SlingshotJump.cs

```diff
@@ -0,0 +1,311 @@
+﻿// Slingshot Jump|Locomotion|20120
+namespace VRTK
+{
+    using System;
+    using GrabAttachMechanics;
+    using UnityEngine;
+
+    /// <summary>
+    /// Event Payload
+    /// </summary>
+    /// <param name="target">The GameObject of the interactable object that is being interacted with by the controller.</param>
+    public struct SlingshotJumpEventArgs
+    {
+        public GameObject target;
+    }
+
+    /// <summary>
+    /// Event Payload
+    /// </summary>
+    /// <param name="sender">this object</param>
+    /// <param name="e"><see cref="PlayerClimbEventArgs"/></param>
+    public delegate void SlingshotJumpEventHandler(object sender, SlingshotJumpEventArgs e);
+
+    /// <summary>
+    /// Slingshot Jump allows player jumping based on the direction and amount pulled back of each controller. This slingshots the player in defined direction and speed.
+    /// </summary>
+    /// <example>
+    /// `VRTK/Examples/037_CameraRig_ClimbingFalling` shows how to set up a scene with slingshot jumping. This script just needs to be added to the PlayArea object and the requested forces and buttons set.
+    /// </example>
+    [AddComponentMenu("VRTK/Scripts/Locomotion/VRTK_SlingshotJump")]
+    public class VRTK_SlingshotJump : MonoBehaviour
+    {
+        [Header("SlingshotJump Settings")]
+
+        [Tooltip("How close together the trigger releases have to be to initiate a jump.")]
+        public float releaseWindowTime = 0.5f;
+        [Tooltip("Multiplier that increases the jump strength.")]
+        public float velocityMultiplier = 5.0f;
+        [Tooltip("The maximum velocity a jump can be.")]
+        public float velocityMax = 8.0f;
+
+        [Tooltip("This button will cancel an already tensioned sling shot.")]
+        [SerializeField]
+        protected VRTK_ControllerEvents.ButtonAlias cancelButton = VRTK_ControllerEvents.ButtonAlias.Undefined;
```

Comment:
Hmmm, good point. I was thinking of using properties, but those don't show up in the inspector. I can add some Get/Set functions instead. I just want to avoid people setting these directly and wondering why it's not working.